node/src/components/storage.rs

```diff
@@ -556,7 +556,22 @@ impl Storage {
         match incoming.message {
             NetRequest::Deploy(ref serialized_id) => {
                 let id = decode_item_id::<Deploy>(serialized_id)?;
-                let opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+                let mut opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+
+                if let Some(deploy) = &mut opt_item {
+                    let mut txn = self
+                        .env
+                        .begin_ro_txn()
+                        .map_err(|e| FatalStorageError::InternalStorage(e.into()))?;
+                    if let Some(finalized_approvals) = txn
+                        .get_value(self.finalized_approvals_db, &id)
+                        .map_err(|e| {
+                            GetRequestError::Fatal(FatalStorageError::InternalStorage(e))
+                        })?
+                    {
+                        deploy.replace_approvals(finalized_approvals);
+                    }
```

Comment:
I can see why `NetRequest::BlockAndDeploys` now adds finalized approvals - b/c it's used in the `chain_synchronizer/operations.rs` when syncing but where in the syncing process do we make use of the fact that also `NetResponse::Deploy` will include the finalized approvals?

Reply:
We don't, but Fraser suggested I make the change to this `NetRequest` as well.